src/main/assets/js/session-timeout.ts

```diff
@@ -0,0 +1,41 @@
+import _ from 'lodash';
+
+const eventThrottleTimer = 5 * 60 * 1000; // 5 minutes
+const sessionTimeoutInterval = 20 * 60 * 1000; // 20 minutes
+let timeout;
+
+const saveBeforeSessionTimeout = () => {
+  const form = document.getElementById('main-form');
+  const formData = new FormData(form as HTMLFormElement);
+  formData.append('saveBeforeSessionTimeout', 'true');
+  const url = window.location.pathname;
+  const body = Object.fromEntries(formData);
+  fetch(url, {
+    method: 'POST',
+    headers: {
+      'Content-type': 'application/json',
+      'csrf-token': body._csrf as string,
+    },
+    body: JSON.stringify(body),
+  }).then(response => {
```

Comment:
maybe it would be better to make this async and use await?

Reply:
just for my understanding, why would that be better? i dont have a preference really but just thought the .then block makes it easier to read and as its combined together.

Reply:
No technical difference, just personal preference I guess. I prefer to stick to the `const someVariable = something()` look.

For example:

```
  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-type': 'application/json',
      'csrf-token': body._csrf as string,
    },
    body: JSON.stringify(body),
  });

  if (response.redirected) {
    window.location.href = response.url;
  }
```